plugins/communications-app/TaskAlertModalForm/index.jsx

```diff
@@ -0,0 +1,163 @@
+import { useSelector, useDispatch } from '@communications-app/src/components/bulk-email-tool/bulk-email-form/BuildEmailFormExtensible/context';
+import { actionCreators as formActions } from '@communications-app/src/components/bulk-email-tool/bulk-email-form/BuildEmailFormExtensible/context/reducer';
+
+import { postBulkEmailInstructorTask, patchScheduledBulkEmailInstructorTask } from './api';
+import { AlertMessage, EditMessage } from './AlertTypes';
+
+const TaskAlertModalForm = ({
+  courseId,
+  isTaskAlertOpen,
+  closeTaskAlert,
+}) => {
+  const [, dispatch] = useContext(BulkEmailContext);
+  const formData = useSelector((state) => state.form);
+  const dispatchForm = useDispatch();
+
+  const {
+    isScheduled,
+    emailRecipients,
+    scheduleDate = '',
+    scheduleTime = '',
+    isEditMode = false,
+    subject,
+    emailId = '',
+    schedulingId = '',
+    body,
+    isScheduleButtonClicked = false,
+    isFormSubmitted = false,
+  } = formData;
+
+  const changeFormStatus = (status) => dispatchForm(formActions.updateForm({ formStatus: status }));
+  const handleResetFormValues = () => dispatchForm(formActions.resetForm());
+
+  const handlePostEmailTask = async () => {
+    const emailData = new FormData();
+    emailData.append('action', 'send');
+    emailData.append('send_to', JSON.stringify(emailRecipients));
+    emailData.append('subject', subject);
+    emailData.append('message', body);
+
+    if (isScheduled) {
+      emailData.append('schedule', new Date(`${scheduleDate} ${scheduleTime}`).toISOString());
+    }
+
+    changeFormStatus('pending');
+
+    try {
+      await postBulkEmailInstructorTask(emailData, courseId);
+      const newFormStatus = isScheduled ? 'completeSchedule' : 'complete';
+      changeFormStatus(newFormStatus);
+      setTimeout(() => handleResetFormValues(), 3000);
+    } catch {
+      changeFormStatus('error');
+    }
+  };
+
+  const handlePatchEmailTask = async () => {
+    const emailRecipientsValue = emailRecipients;
+    const emailSubject = subject;
+    const emailBody = body;
+
+    const emailData = {
+      email: {
+        targets: emailRecipientsValue,
+        subject: emailSubject,
+        message: emailBody,
+        id: emailId,
+      },
+      schedule: isScheduled ? new Date(`${scheduleDate} ${scheduleTime}`).toISOString() : null,
+    };
+
+    changeFormStatus('pending');
+
+    try {
+      await patchScheduledBulkEmailInstructorTask(emailData, courseId, schedulingId);
+      changeFormStatus('completeSchedule');
+      setTimeout(() => handleResetFormValues(), 3000);
+    } catch {
+      changeFormStatus('error');
+    }
+  };
+
+  const createEmailTask = async () => {
+    const isScheduleValid = isScheduled ? scheduleDate.length > 0 && scheduleTime.length > 0 : true;
+    const isFormValid = emailRecipients.length > 0 && subject.length > 0
+    && body.length > 0 && isScheduleValid;
```

Comment:
Same note as the one I left on `BodyForm` about having validation code in multiple places. I'd prefer to not require anyone in the future that updates the validation logic to make changes in more than one place.